src/test/java/com/atlassian/db/replica/it/DualConnectionIT.java

```diff
@@ -79,6 +79,22 @@ public void shouldRunNextValOnMainDatabase() throws SQLException {
         }
     }
 
+    @Test
+    public void shouldChangeTransactionIsolationLevel() throws SQLException {
+        try (PostgresConnectionProvider connectionProvider = new PostgresConnectionProvider()) {
+            final Connection connection = DualConnection.builder(
+                connectionProvider,
+                CircularConsistency.permanentConsistency().build()
+            ).build();
+
+            connection.setTransactionIsolation(Connection.TRANSACTION_READ_UNCOMMITTED);
+            connection.setAutoCommit(false);
+            connection.prepareStatement("SELECT 1;").executeQuery();
+            connection.prepareStatement("SELECT 1;").executeQuery();
+            connection.commit();
+        }
```

Comment:
Does this test how the connection tried to be initialized twice with changed parameters?

Reply:
Yes.